sycl/include/CL/sycl/accessor.hpp

```diff
@@ -1329,6 +1551,221 @@ class accessor<DataT, Dimensions, AccessMode, access::target::image_array,
   }
 };
 
+template <typename DataT, int Dimensions = 1,
+          access_mode AccessMode = access_mode::read_write>
+class host_accessor
+    : public accessor<DataT, Dimensions, AccessMode, target::host_buffer,
+                      access::placeholder::false_t> {
+protected:
+  using AccessorT = accessor<DataT, Dimensions, AccessMode, target::host_buffer,
+                             access::placeholder::false_t>;
+
+  constexpr static int AdjustedDim = Dimensions == 0 ? 1 : Dimensions;
+
+  template <typename T, int Dims> static constexpr bool IsSameAsBuffer() {
+    return std::is_same<T, DataT>::value && (Dims > 0) && (Dims == Dimensions);
+  }
+
+#if __cplusplus > 201402L
+
+  template <typename TagT> static constexpr bool IsValidTag() {
+    return std::is_same<TagT, mode_tag_t<AccessMode>>::value;
+  }
+
+#endif
+
+  void
+  __init(typename accessor<DataT, Dimensions, AccessMode, target::host_buffer,
+                           access::placeholder::false_t>::ConcreteASPtrType Ptr,
+         range<AdjustedDim> AccessRange, range<AdjustedDim> MemRange,
+         id<AdjustedDim> Offset) {
+    AccessorT::__init(Ptr, AccessRange, MemRange, Offset);
+  }
+
+public:
+  host_accessor() : AccessorT() {}
+
+  // The list of host_accessor constructors with their arguments
+  // -------+---------+-------+----+----------+--------------
+  // Dimensions = 0
+  // -------+---------+-------+----+----------+--------------
+  // buffer |         |       |    |          | property_list
+  // buffer | handler |       |    |          | property_list
+  // -------+---------+-------+----+----------+--------------
+  // Dimensions >= 1
+  // -------+---------+-------+----+----------+--------------
+  // buffer |         |       |    |          | property_list
+  // buffer |         |       |    | mode_tag | property_list
+  // buffer | handler |       |    |          | property_list
```

Comment:
Why are there `host_accessor` constructors that accept a `handler`?

Reply:
This is a placeholder for future extensions to support host tasks. [Host tasks proposal](https://github.com/codeplaysoftware/standards-proposals/blob/master/host_task/host_task.md). We will need blocking and non-blocking accessors. Handler will destinguish non-blocking accessor for their typical behavior.

At the moment their behavior is the same, though. Additional internal restructuring will be required to support non-blocking host accessors.

Reply:
I'm fine with keeping those as a placeholder, but since they currently have invalid behaviour, it should be forbidden to call them, e. g. via `static_assert`. I'd also appreciate mentioning these are just placeholders in either a comment or the static_assert message.

Reply:
Removed placeholder for the time being, and left corresponding comment in the footer of this table